### PR TITLE
Add PVC to object status

### LINKED ISSUE #2641: Unbound PVC still shown as "OK"
**What steps did you take and what happened:**
[A clear and concise description of what the bug is, and what commands you ran.)

I deployed a workload in a local cluster and introduced an error: the PVC spec calls a Storage Class that doesn't exist. The workload is not working, the PVC is not bound but Octant shows it in a "OK" state which can be confusing for troubleshooting.

image

**What did you expect to happen:**
Octant should reflect the accurate status of a PVC
**Anything else you would like to add:**

**Environment:**

- Octant version (use `octant version`): 0.21.0
- Kubernetes version (use `kubectl version`): 1.21
- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc): macOS 11.3.1

### PR DESCRIPTION
**Pull request by ftovaro**
Signed-off-by: ftovaro <f.tovar12@gmail.com>

**What this PR does / why we need it**:
We where having a bug in the status of a PVC, displaying that the PVC was `Ok` when it was `Pending` due to an error in the config file

image

**Which issue(s) this PR fixes**
- Fixes #2641 

**Special notes for your reviewer**:

**Release note**:
```
release-note
```

### CODE REVIEW

File: internal/objectstatus/persistent_volume_claim.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+	if pvc.Status.Phase == "Pending" {
+		return ObjectStatus{
+			NodeStatus: component.NodeStatusWarning,
+			Details:    []component.Component{component.NewText("PVC cannot be found")}}, nil
+	}
+
+	return ObjectStatus{
+		NodeStatus: component.NodeStatusOK,
+		Details:    []component.Component{component.NewText("v1 PersistentVolume is OK")},
```
Comment (GuessWhoSamFoo) [inline internal/objectstatus/persistent_volume_claim.go:42]: This is the status of a PVC, not PV

File: internal/objectstatus/persistent_volume_claim_test.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
+/*
+Copyright (c) 2019 the Octant contributors. All Rights Reserved.
```
Comment (GuessWhoSamFoo) [inline internal/objectstatus/persistent_volume_claim_test.go:2]: nit: 2021

File: internal/objectstatus/persistent_volume_claim.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import (
+	"context"
+
+	"github.com/vmware-tanzu/octant/pkg/view/component"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"github.com/pkg/errors"
```
Comment (wwitzel3) [inline internal/objectstatus/persistent_volume_claim.go:16]: sort imports: built-in, external, internal

File: internal/objectstatus/persistent_volume_claim.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+func persistentVolumeClaim(_ context.Context, object runtime.Object, _ store.Store, _ link.Interface) (ObjectStatus, error) {
+	if object == nil {
+		return ObjectStatus{}, errors.Errorf("cronjob is nil")
+	}
+
+	pvc := &corev1.PersistentVolumeClaim{}
+
+	if err := scheme.Scheme.Convert(object, pvc, 0); err != nil {
+		return ObjectStatus{}, errors.Wrap(err, "convert object to v1 PersistentVolumeClaim")
```
Comment (wwitzel3) [inline internal/objectstatus/persistent_volume_claim.go:33]: Use `fmt.Errorf` with `%w`

File: internal/objectstatus/persistent_volume_claim.go (hunk 3 above)
Comment (wwitzel3) [inline internal/objectstatus/persistent_volume_claim.go:16]: we are moving away from the errors package

File: internal/objectstatus/persistent_volume_claim.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"github.com/vmware-tanzu/octant/internal/link"
+	"github.com/vmware-tanzu/octant/pkg/store"
+	"github.com/vmware-tanzu/octant/pkg/view/component"
+)
+
+func persistentVolumeClaim(_ context.Context, object runtime.Object, _ store.Store, _ link.Interface) (ObjectStatus, error) {
+	if object == nil {
+		return ObjectStatus{}, errors.New("cronjob is nil")
```
Comment (wwitzel3) [inline internal/objectstatus/persistent_volume_claim.go:24]: Fix copy/paste error message.

**Review by wwitzel3 — APPROVED** [review]

**Pull request merged**